package/libs/openssl/patches/110-openwrt_targets.patch

```diff
@@ -0,0 +1,26 @@
+--- /dev/null
++++ b/Configurations/25-openwrt.conf
+@@ -0,0 +1,23 @@
++%targets = (
++    "linux-x86_64-openwrt" => {
++        inherit_from    => [ "linux-x86_64" ],
++	cflags		=> add("\$(OPENWRT_OPTIMIZATION_FLAGS)"),
++    },
++    "linux-aarch64-openwrt" => {
++        inherit_from    => [ "linux-aarch64" ],
++	cflags		=> add("\$(OPENWRT_OPTIMIZATION_FLAGS)"),
++    },
++    "linux-armv4-openwrt" => {
++        inherit_from    => [ "linux-armv4" ],
++	cflags		=> add("\$(OPENWRT_OPTIMIZATION_FLAGS)"),
++    },
++    "linux-mips32-openwrt" => {
++        inherit_from    => [ "linux-mips32" ],
++	cflags		=> add("\$(OPENWRT_OPTIMIZATION_FLAGS)"),
++    },
++    "linux-generic-openwrt" => {
++        inherit_from    => [ "linux-generic" ],
```

Comment:
@cotequeiroz 
This should probably say ```linux-generic32``` as there's no linux-generic :-)
Also, Octeon (mips64) should have it's own entry
https://github.com/openssl/openssl/blob/OpenSSL_1_1_0-stable/Configurations/10-main.conf#L722

Reply:
Thanks for pointing this out!  I'm going ahead and add all of the currently supported archs.  I'm compiling-testing each arch, and I'll add the commit when I'm done.  `arc` is the only one falling back to `generic32`.  I can't run test most them, though.  Just to be clear, I'm testing one of each `Target-Arch:` entry from `tmp/.targetinfo` (i.e. aarch64, arc, arm, armeb, i386, mips, mips64, mips64el, mipsel, powerpc, x86_64), not every `Target:` entry.

Reply:
@cotequeiroz 
https://github.com/diizzyy/openwrt/commit/405cae4c54d3c06da26914ecc16826f087cd8386
Ugly hack (lazy) but you get the idea :)

Reply:
I'm still not quite done yet.  It was good to test it, 'cause I picked up some more building problems (like openssl installing to lib64, and the openwrt Makefile copying files from lib, in mips64).  Right now I'm testing if it's OK to not add `no-sse2` to every arch that does not support.  It should be safe, but I'm afraid to break things, so I try not to touch too many things at once.